src/main/java/net/imglib2/img/basictypelongaccess/unsafe/NativeBooleanUnsafe.java

```diff
@@ -0,0 +1,105 @@
+/*-
+ * #%L
+ * ImgLib2 data structures using Unsafe.
+ * %%
+ * Copyright (C) 2017 - 2021 Howard Hughes Medical Institute.
+ * %%
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright notice,
+ *    this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright notice,
+ *    this list of conditions and the following disclaimer in the documentation
+ *    and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDERS OR CONTRIBUTORS BE
+ * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ * #L%
+ */
+package net.imglib2.img.basictypelongaccess.unsafe;
+
+import net.imglib2.img.basictypeaccess.volatiles.VolatileBooleanAccess;
+import net.imglib2.img.basictypelongaccess.BooleanLongAccess;
+
+public class NativeBooleanUnsafe
```

Comment:
This class should be named just `BooleanUnsafe`, not `NativeBooleanUnsafe`. The term `Native` in the context of ImgLib2 means "backed by a Java primitive type", and this class is not backed by `boolean[]`; it is backed by `ByteUnsafe` under the hood. It's really about providing a boolean-centric view into direct/unsafe memory access.

In ImgLib2, there is the `NativeBoolType`, but it's named that because that interface extends both `BooleanType` and `NativeType` so that it can be used with the performant `Img` implementations that extend `AbstractNativeImg`.

In this repository, classes have the word `Native` only to extend core ImgLib2's codebase from `int` indices to `long`. So we have the direct correlation:

* `AbstractNativeImg` &rarr; `AbstractNativeLongAccessImg`
* `NativeImg` &rarr; `NativeLongAccessImg`
* `NativeImgFactory` &rarr; `NativeLongAccessImgFactory`
* `AbstractNativeType` &rarr; `AbstractNativeLongAccessType`
* `NativeType` &rarr; `NativeLongAccessType`
* `NativeTypeFactory` &rarr; `NativeLongAccessTypeFactory`

Arguably, the `NativeLongAccessType` classes here should not have been named that way, because they are no longer backed by Java primitive arrays, but rather by unsafe memory addresses. (If you check that class, you can see even the javadoc was copy-pasted and refers to Java primitive arrays, erroneously now.) Note that the concrete implementations such as `GenericIntLongAccessType` are again named in accordance with ImgLib2 core, with `NativeAccess` sandwiched in.

So then for consistency with ImgLib2 core and boolean types, let's look what boolean types it has?

1. `BooleanType` &ndash; an interface for all boolean-based types &ndash; unfortunately named, but no changing it now
2. `BoolType` &ndash; a type for wrapping singleton `boolean` values, which unfortunately does not implement `NativeType`.
3. `NativeBoolType` &ndash; a type that *does* implement `NativeType` and can therefore be used with `ArrayImg` et al. backed by `boolean[]`. It's unfortunate that, unlike the other corresponding primitive-based `NativeType` classes, this one retained `Native` in the name and the others (e.g. `DoubleType`) did not... but it was impossible to name it just `BooleanType` because that already existed as interface (1) above, nor was the name `BoolType` available because it was already used for (2), so some differentiating factor needed to be put into the name, and `NativeBoolType` seemed the least bad/unclear.

So, this third thing `NativeBoolType` is the closest analogue to the `NativeBooleanUnsafe` here, implying this class should be named `NativeBoolUnsafe` for consistency with e.g. `DoubleType` &rarr; `DoubleUnsafe`. But unlike the situation in ImgLib2 core, then name `BooleanUnsafe` is not already taken, and it is more correct in that it is an `Unsafe`-backed access exposing the data as `boolean` values. Therefore, I vote for naming it `BooleanUnsafe`!

Reply:
This all sounds great to me, I'd just appreciate some explanation added to the javadoc explaining this decision, so we remember why this was named as such!

Reply:
My writeup above was much more complex than it needed to be.

The main reason to name this class `BooleanUnsafe` is: it is the `Unsafe` access class implementing `BooleanLongAccess`. Just the same as e.g. `DoubleUnsafe` implements `DoubleLongAccess`. The `Type` implementation names don't even come into it at the `Access` level.